source/server-discovery-and-monitoring/tests/unified/closeInUse-pool-clear.yml

```
@@ -0,0 +1,322 @@
+---
+description: closeInUse
+
+schemaVersion: "1.11"
+
+runOnRequirements:
+  # failCommand appName requirements
+  - minServerVersion: "4.9"
+    serverless: forbid
+    topologies: [ replicaset, sharded, load-balanced ]
+
+createEntities:
+  - client:
+      id: &setupClient setupClient
+      useMultipleMongoses: false
+
+initialData: &initialData
+  - collectionName: &collectionName closeInUse
+    databaseName: &databaseName sdam-tests
+    documents: []
+
+tests:
+  - description: Connection pool clear uses closeInUseConnections=true after monitor timeout
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                useMultipleMongoses: false
+                observeEvents:
+                  - poolClearedEvent
+                  - connectionClosedEvent
+                  - commandStartedEvent
+                  - commandSucceededEvent
+                  - commandFailedEvent
+                  - connectionCheckedOutEvent
+                  - connectionCheckedInEvent
+                uriOptions:
+                  connectTimeoutMS: 500
+                  heartbeatFrequencyMS: 500
+                  appname: closeInUse
+                  retryReads: false
+                  minPoolSize: 0
+            - database:
+                id: &database database
+                client: *client
+                databaseName: *databaseName
+            - collection:
+                id: &collection collection
+                database: *database
+                collectionName: *collectionName
+            - thread:
+                id: &thread1 thread1
+      - name: insertOne
+        object: *collection
+        arguments:
+          document:
+            _id: 1
+      # simulate a long-running query
+      - name: runOnThread
+        object: testRunner
+        arguments:
+          thread: *thread1
+          operation:
+            name: find
+            object: *collection
+            arguments:
+              filter:
+                $where : sleep(5000) || true
+            expectError:
+              isError: true
+        # Configure the monitor check to fail with a timeout.
+        # Use times: 2 so that the RTT hello is blocked as well.
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *setupClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode:
+              times: 2
+            data:
+              failCommands:
+                - hello
+                - isMaster
+              blockConnection:  true,
+              blockTimeMS: 1500,                
+              appName: closeInUse        
+      - name: waitForThread
+        object: testRunner,
+        arguments: {
+          "thread": "thread1"
+        }
```

For consistency, don't use braces to demarcate a document unless you're writing it out on a single line. Single-line notation typically isn't done for `arguments` fields, so I wouldn't suggest doing that here.

Also, there's no need to quote the key and you should refer back to the existing anchor your defined instead of repeating "thread1" as a string literal.

```suggestion
        arguments:
          thread: *thread1
```